### DISCUSSION TITLE
Absence of sugar readings in the History section if one-minute values are received from the sensor

### DISCUSSION THREAD

**Discussion by Inkalika** [NightscoutFoundation/xDrip#3190] [Bug reports] [answered]
xDrip version 23/09/23
I came across the fact that in the History section there is a limit on the number of sugar values. For example, if I receive 1 minute readings from Juggluco (Libre2), then for part of the night from 00:00 to 03:00-06:00 am (depending on how many readings there were during the day) there will be no readings. That is, the readings per day will be less than 1х60х24.
If I select 5-minute sugar values for display, then the sugar readings for the day will be complete.
Is it possible to increase the number of sugar storage points in the History section?

**Comment by Navid200** [comment]
@tzachi-dar Is there a hardcode that needs to be updated now that there are 1-reading per minute options appearing?

**Reply by ah7675** [reply]
Yes, there is.
"LIMIT" when query from database.
I couldn't recall the number currently xDrip used but it must be less than 1440.
There are 1440 minutes for each day whereas xDrip assumes that there are no more than (1440/5) = 288 data points.

**Reply by ah7675** [reply]
https://github.com/NightscoutFoundation/xDrip/blob/c16498075051bcb0a71f573dc0b81f4c39bb5eeb/app/src/main/java/com/eveningoutpost/dexdrip/BGHistory.java#L151-L152

**Comment by Navid200** [comment]
@Inkalika Is there an xDrip log you see when this happens?

**Reply by Inkalika** [reply]
@Navid200
No. When viewing the History, I saw that the days when xdrip collected 1-minute readings, there is no part of the readings at the beginning of the night and so every day. Since this is important information, I turned off the 1-minute and left only 5 minutes.

**Reply by Inkalika** [reply]
sent a log from form xdrip

**Comment by Navid200** [comment]
@Inkalika We have just had a Nightly release: https://github.com/NightscoutFoundation/xDrip/releases/tag/2023.11.29

It increases the capacity of the daily history so that even if you have one reading per minute, it will maintain all of them.
Please give it a try and let me know how it works.

**Reply by Inkalika** [reply]
Ok. 
I will test it for a couple of days and give feedback

**Reply by Inkalika** [reply]
It's work. Thx

**Reply by Navid200** [reply, marked as answer]
Great.  Thanks for the feedback

**Reply by Inkalika** [reply]
I don’t know how it was before, but it seems to me that it wasn’t like that.  the follower works like xdrip sync, after turning on the Internet and turning it on a few hours later, therapy for the hours when the follower was turned off does not load (even if the Internet was turned off for an hour), the glucose level is also not loaded for the entire period of shutdown

I also noticed that the follower (xdrip sync) does not receive the sugar value every 1 minute, but once every 3-4 minutes.  Nightscout receives every 1-minute

PS: looked in history.  Yes, in the version dated 09/23/23, after turning on the Internet, therapy was loaded for all the clocks that were turned off and all glucose knowledge.  Then I had 5-minute glucose values ​​turned on

PPS: restored the old version on the follower.  I left a new version on the master.  I’ll see how this combination will restore therapy and sugar levels after the Internet is restored

**Reply by Inkalika** [reply]
changing the version of the program on the follower did not help.  after the Internet was restored, therapy was not counted for the period of no Internet

**Reply by Navid200** [reply]
Unless you are sure this is related to the previous issue, it may be better to open a new discussion.

If you ever suspect that an update to xDrip may have caused a problem, you can easily install an older version of xDrip to verify.  To do that, you can just install the older version over your current version.  There is no need to uninstall first.

If there is no relationship between the two problems, there is no benefit in using the same description.  In fact, it could be confusing to do that.
